metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java

```diff
@@ -216,28 +215,75 @@ public Integer count(final QualifiedName name) {
      */
     @Override
     public PartitionsSaveResponseDto save(final QualifiedName name, final PartitionsSaveRequestDto dto) {
-        PartitionsSaveResponseDto result = new PartitionsSaveResponseDto();
         final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
-        final ConnectorRequestContext connectorRequestContext = converterUtil.toConnectorContext(metacatRequestContext);
         final ConnectorPartitionService service = connectorManager.getPartitionService(name.getCatalogName());
         final List<PartitionDto> partitionDtos = dto.getPartitions();
         // If no partitions are passed, then return
         if (partitionDtos == null || partitionDtos.isEmpty()) {
-            return result;
+            return new PartitionsSaveResponseDto();
         }
-        final List<String> partitionIdsForDeletes = dto.getPartitionIdsForDeletes();
-        registry.gauge(this.partitionAddGaugeId.withTags(name.parts()), partitionDtos.size());
-        registry.distributionSummary(
-            this.partitionAddDistSummary.withTags(name.parts())).record(partitionDtos.size());
-
         if (!tableService.exists(name)) {
             throw new TableNotFoundException(name);
         }
+        //optimization for metadata only updates (e.g. squirrel) , assuming only validate partitions are requested
+        if (dto.getSaveMetadataOnly()) {
+            return savePartitionMetadataOnly(metacatRequestContext, dto, name, partitionDtos);
+        }
```

Comment:
Can you do an else to make a clear separation.

Reply:
Sure. will do.